### ISSUE TITLE
Unable to find element by exact matching inner text

### ISSUE THREAD

**Issue by Dmitry-Ostashev** [DevExpress/testcafe#1292]
### Are you requesting a feature or reporting a bug?

feature

### What is the current behavior?

Selector.withText(text)	method creates a selector that filters a matching set by a containing the specified text. 

### What is the expected behavior?

Sometimes it is necessary to find element by exact matching inner text

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js

```

### Specify your

* operating system:
* testcafe version:
* node.js version:

**Comment by AndreyBelym** [comment]
Hello, Dmitry! I guess, you want something like Selector('*').withText(theText)`. It matches every element on the page, and then filters it with the specified text.

**Comment by Dmitry-Ostashev** [comment]
I need a strong match, that means "equal", not "contains". 

For example, a have two divs "Object 1" and "Object 11". I want to find only "Object 1", but with .withText('Object 1') I will get both these divs.

I have solved this problem by the following approach:

`var innerTextEquals = ClientFunction((element, text) => {	
		return $.trim(element.innerText) === $.trim(text);
});

//...
Selector('.dx-navigation-item').filter((node, idx) => { 
	return innerTextEquals(node, itemCaption)
}, { innerTextEquals, itemCaption });`

**Comment by inikulin** [comment]
I guess it would be nice to have this feature. As a temporary workaround you can represent your text as regexp and set boundaries:
```js
import escapeRegExp from 'lodash/escapeRegExp';

function getWholeTextRe(text) {
    return new RegExp(`^${escapeRegExp(text)}$`);
}

// use it as t.expect(smthg).contains(getWholeTextRe('myText'));
```

**Comment by inikulin** [comment]
We need to figure out API for that. How about `Selector.withWholeText()`? \cc @DevExpress/testcafe @VasilyStrelyaev

**Comment by VasilyStrelyaev** [comment]
what about `withTextStrictly` or `withStrictText`

**Comment by helen-dikareva** [comment]
May be `withExactText`

**Comment by qaiser110** [comment]
May be better to change `withText` to do exact search by default

and add a new method `withTextContains` or `withTextIncludes` or `withPartialText`

**Comment by AlexanderMoskovkin** [comment]
I wouldn't like to make a breaking change here. It can break a lot of existing tests. The `withText` method can take a regexp so it's quite flexible for any cases. So I'd like to leave it as is

**Comment by AlexanderMoskovkin** [comment]
closed in #2124
